test/test_retry.py

```diff
@@ -131,6 +131,18 @@ def test_zero_backoff(self):
         retry = retry.increment()
         self.assertEqual(retry.get_backoff_time(), 0)
 
+    def test_backoff_reset_after_redirect(self):
+        retry = Retry(total=100, redirect=5, backoff_factor=0.2)
+        retry = retry.increment()
+        retry = retry.increment()
+        self.assertEqual(retry.get_backoff_time(), 0.4)
+        redirect_response = HTTPResponse(status=302, headers={'location': 'test'})
+        retry = retry.increment(response=redirect_response)
+        self.assertEqual(retry.get_backoff_time(), 0)
+        retry = retry.increment()
+        retry = retry.increment()
+        self.assertEqual(retry.get_backoff_time(), 0.4)
+
```

Comment:
That's fine: feel free to do that. I'd *also* like a check for the backoff function, just because it's one of those implicit things that can easily be overlooked.